dwave/optimization/model.pyx

```diff
@@ -228,6 +228,23 @@ cdef class Model:
         lists = [DisjointList(main, i) for i in range(num_disjoint_lists)]
         return main, lists
 
+    def feasible(self, int index = 0):
+        """Check the feasibility of the state at the input index.
+
+        Args:
+            index: index of the state to check for feasibility.
+
+        Returns:
+            Feasibility of the state.
+        """
+        cdef int num_states = self.state_size()
+        if not -num_states <= index < num_states:
+            raise ValueError(f"index out of range: {index}")
+        elif index < 0:  # allow negative indexing
+            index += num_states
+
+        return self._graph.feasible(self.states._states[index])
```

Comment:
I suspect it seg-faults because you need to initialize the state before calling the feasible method. This should be the cause if in your test, when calling it, you set a state and then called the feasible method right after
So you can add some python tests! :)

Reply:
Sorry, I think I was unclear earlier. I meant here
```python
self._graph.initialize_state(self.states._states[index])
return self._graph.feasible(self.states._states[index])
```

Reply:
Ahh gotcha. Sorry, just my unfamiliarity with the codebase. Anything after initializing the state I'd need to clean up?

Reply:
I think that should be it.. there are two things worth remembering here. The C++ method `initialize_state()` initialize all nodes in the graph (eagerly) to their value given the decision variables. If you were to check every single constraint from the python side, then it would be lazily from checking each constraint. Worth keeping this in mind in case one deals with models with very deep graphs on the objective side and simple constraints on the decisions (kinda a worst case scenario)

Reply:
I'll try doing some performance analysis and see if there's a significant cost with the eager initialization. Also could test with common problem types (i.e. models created by the generator functions) and see if it makes any difference